### PR TITLE
x86: Add disableAVX2/512 options and check XCR0 for OS support

### PR DESCRIPTION
**Pull request by BradleyWood**

### CODE REVIEW

**Comment by BradleyWood** [comment]
jenkins build all

**Comment by BradleyWood** [comment]
FYI, @0xdaryl @JamesKingdon @hzongaro

File: compiler/x/env/OMRCPU.cpp [hunk 1]
```diff
@@ -64,11 +64,37 @@ OMR::X86::CPU::detect(OMRPortLibrary * const omrPortLib)
...
    if (TRUE == omrsysinfo_processor_has_feature(&processorDescription, OMR_FEATURE_X86_OSXSAVE))
       {
-      static const bool disableAVX = feGetEnv("TR_DisableAVX") != NULL;
-      if (((6 & _xgetbv(0)) != 6) || disableAVX) // '6' = mask for XCR0[2:1]='11b' (XMM state and YMM state are enabled)
+      // Check XCRO register for OS support of xmm/ymm/zmm
+      // '6' = mask for XCR0[2:1]='11b' (XMM state and YMM state are enabled)
+      static const bool disableAVX = ((6 & _xgetbv(0)) != 6) || feGetEnv("TR_DisableAVX") != NULL;
+      static const bool disableAVX2 = disableAVX || feGetEnv("TR_DisableAVX2") != NULL;
+      // 'e0' = mask for XCR0[7:5]='111b' (Opmask, ZMM_Hi256, Hi16_ZMM)
+      static const bool disableAVX512 = ((0xe0 & _xgetbv(0)) != 0xe0) || disableAVX || disableAVX2 || feGetEnv("TR_DisableAVX512") != NULL;
```
Comment (0xdaryl) [inline compiler/x/env/OMRCPU.cpp:72]: I realize you checked for XCR0[2:1] on L69 which makes it unnecessary here, but for someone reading the Intel spec and expecting to find those bits checked in the AVX512 case they'll have to study this code to grok this fact.

It costs nothing to re-check them here, so for readability can you change the mask to `0xe6` (I think)?  Unless there is some other reason you didn't do that, in which case please enlighten.

Ditto for the duplicate code in the JIT.

Comment (0xdaryl) [reply compiler/x/env/OMRCPU.cpp:72]: Thanks for making the change, but is there a reason you didn't update this mask?

**Comment by 0xdaryl** [comment]
Jenkins build all

**Comment by 0xdaryl** [comment]
Added WIP to the title.

**Comment by BradleyWood** [comment]
Jenkins build all

**Comment by BradleyWood** [comment]
I found an issue with TR_DisableAVX in openJ9 without this PR. I am investigating.

**Comment by BradleyWood** [comment]
jenkins build all

**Comment by BradleyWood** [comment]
@0xdaryl Can you take another review of this PR. I expect it will fix eclipse-openj9/openj9#19408 and customer issue. Would be nice to get those fixed for 0.49, if possible.

**Comment by BradleyWood** [comment]
This PR shouldn't break J9. However, J9 doesn't respect when OMR disables features.  eclipse-openj9/openj9#20691 will be required to use any of these options.

**Comment by BradleyWood** [comment]
jenkins build all

**Review by 0xdaryl — APPROVED** [review]

**Comment by 0xdaryl** [comment]
@BradleyWood : in case it wasn't apparent, while you were away this PR was reverted because of Windows compile failures.

The issue was with the undefined `compilation` symbol in the `OMR::X86::CPU::is_feature_disabled(uint32_t feature)` function.  Given that a CPU class is attached to every compilation object now, it makes sense to cache the compilation object in the CPU class if it exists for just these kinds of checks.  Of course, a NULL compilation object is still a possibility if this class is used outside of the JIT or in early/late initialization.  However, I'll let you consider the alternatives for how to resolve.

**Comment by 0xdaryl** [comment]
@BradleyWood : are you planning on re-introducing this PR with a fix?

**Pull request merged**